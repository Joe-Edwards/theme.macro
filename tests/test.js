const path = require('path');
const pluginTester = require('babel-plugin-tester').default;
const plugin = require('babel-plugin-macros');

pluginTester({
  plugin,
  babelOptions: { filename: __filename },
  title: 'Theme Macro',
  tests: [
    { title: 'Simple', fixture: path.join(__dirname, '__fixtures__', 'simple.js'), snapshot: true, },
    { title: 'Props', fixture: path.join(__dirname, '__fixtures__', 'props.js'), snapshot: true, },
    { title: 'Nesting', fixture: path.join(__dirname, '__fixtures__', 'nested.js'), snapshot: true, },
    { title: 'Different usages', fixture: path.join(__dirname, '__fixtures__', 'usage.js'), snapshot: true, },
    { title: 'Shadowing', fixture: path.join(__dirname, '__fixtures__', 'shadowing.js'), snapshot: true, },
    { title: 'Outside a template literal', fixture: path.join(__dirname, '__fixtures__', 'error.js'), error: 'The theme macro at line 3 is not used within a tagged template literal' },
    { title: 'In an untagged template literal', fixture: path.join(__dirname, '__fixtures__', 'error2.js'), error: 'The theme macro at line 3 is not used within a tagged template literal' }
  ],
});
