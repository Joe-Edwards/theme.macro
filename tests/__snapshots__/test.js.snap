// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Macro Different usages: Different usages 1`] = `

import theme from '../../theme.macro';

css\`\${theme.foo}\`
styled.div\`\${theme.foo}\`
styled(Foo)\`\${theme.foo}\`

      ↓ ↓ ↓ ↓ ↓ ↓

css\`
  \${(_props) => _props.theme.foo}
\`;
styled.div\`
  \${(_props2) => _props2.theme.foo}
\`;
styled(Foo)\`
  \${(_props3) => _props3.theme.foo}
\`;


`;

exports[`Theme Macro Nesting: Nesting 1`] = `

import theme from '../../theme.macro';

css\`\${props => props.foo && css\`\${theme.bar}\`}\`
css\`\${props => props.foo ? css\`\${theme.bar}\` : css\`\${theme.baz}\`}\`
css\`\${theme.foo ? css\`\${theme.bar}\` : css\`\${theme.baz}\`}\`
css\`\${theme.foo ? \`\${theme.bar}\` : \`\${theme.baz}\`}\`

      ↓ ↓ ↓ ↓ ↓ ↓

css\`
  \${(props) =>
    props.foo &&
    css\`
      \${(_props) => _props.theme.bar}
    \`}
\`;
css\`
  \${(props) =>
    props.foo
      ? css\`
          \${(_props2) => _props2.theme.bar}
        \`
      : css\`
          \${(_props3) => _props3.theme.baz}
        \`}
\`;
css\`
  \${(_props4) =>
    _props4.theme.foo
      ? css\`
          \${(_props5) => _props5.theme.bar}
        \`
      : css\`
          \${(_props6) => _props6.theme.baz}
        \`}
\`;
css\`
  \${(_props7) =>
    _props7.theme.foo ? \`\${_props7.theme.bar}\` : \`\${_props7.theme.baz}\`}
\`;


`;

exports[`Theme Macro Props: Props 1`] = `

import theme from '../../theme.macro';

const computed = 'theme';

css\`\${() => theme.foo}\`
css\`\${props => theme.foo + props.bar}\`
css\`\${function (props) { theme.foo + props.bar }}\`
css\`\${({ bar }) => theme.foo + bar}\`
css\`\${({ bar: { baz } }) => theme.foo + baz}\`
css\`\${({ '-bar-': bar }) => theme.foo + bar}\`
css\`\${({ bar, theme: t }) => theme.foo + bar + t.baz}\`
css\`\${({ bar, 'theme': t }) => theme.foo + bar + t.baz}\`
css\`\${({ bar, ...rest }) => theme.foo + bar + rest.baz}\`
css\`\${({ [computed]: bar }) => theme.foo + bar}\`

      ↓ ↓ ↓ ↓ ↓ ↓

const computed = "theme";
css\`
  \${(_props) => _props.theme.foo}
\`;
css\`
  \${(props) => props.theme.foo + props.bar}
\`;
css\`
  \${function (props) {
    props.theme.foo + props.bar;
  }}
\`;
css\`
  \${({ bar, theme: _theme }) => _theme.foo + bar}
\`;
css\`
  \${({ bar: { baz }, theme: _theme2 }) => _theme2.foo + baz}
\`;
css\`
  \${({ "-bar-": bar, theme: _theme3 }) => _theme3.foo + bar}
\`;
css\`
  \${({ bar, theme: t }) => t.foo + bar + t.baz}
\`;
css\`
  \${({ bar, theme: t, theme: _theme4 }) => _theme4.foo + bar + t.baz}
\`;
css\`
  \${(_props2) => ({ bar, ...rest }) => _props2.theme.foo + bar + rest.baz}
\`;
css\`
  \${({ [computed]: bar, theme: _theme5 }) => _theme5.foo + bar}
\`;


`;

exports[`Theme Macro Shadowing: Shadowing 1`] = `

import theme from '../../theme.macro';

css\`\${(props) => theme.foo + props.bar + ((props) => theme.baz)()}\`
css\`\${({ theme: t }) => theme.foo + t.bar + ((t) => theme.baz)()}\`

      ↓ ↓ ↓ ↓ ↓ ↓

css\`
  \${(_props) => _props.theme.foo + _props.bar + ((props) => _props.theme.baz)()}
\`;
css\`
  \${({ theme: _t }) => _t.foo + _t.bar + ((t) => _t.baz)()}
\`;


`;

exports[`Theme Macro Simple: Simple 1`] = `

import theme from '../../theme.macro';

css\`\${theme.foo}\`
css\`\${theme.foo.bar}\`
css\`\${theme.foo[8] + 7}\`
css\`\${bar(theme.foo)}\`
css\`\${theme.foo + theme.bar}\`

      ↓ ↓ ↓ ↓ ↓ ↓

css\`
  \${(_props) => _props.theme.foo}
\`;
css\`
  \${(_props2) => _props2.theme.foo.bar}
\`;
css\`
  \${(_props3) => _props3.theme.foo[8] + 7}
\`;
css\`
  \${(_props4) => bar(_props4.theme.foo)}
\`;
css\`
  \${(_props5) => _props5.theme.foo + _props5.theme.bar}
\`;


`;
